client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -606,14 +627,45 @@ export default {
           const mergedPage = new compose.Page({ namespaceID, ...page, blocks: this.blocks })
 
           this.updatePage(mergedPage).then((page) => {
-            this.unsavedBlocks.clear()
-            this.toastSuccess(this.$t('notification:page.saved'))
-            if (closeOnSuccess) {
-              this.$router.push({ name: 'admin.pages' })
-            } else if (previewOnSuccess) {
-              this.$router.push({ name: 'page', params: { pageID: this.pageID } })
+            const hasTempID = this.page.blocks.filter(b => b.kind === 'Tabs').some(b => b.options.tabs.some(t => t.blockID.length > 8))
+            if (hasTempID) {
+              const tabsWithTempIDs = this.page.blocks.filter(b => b.kind === 'Tabs').filter(b => b.options.tabs.some(t => t.blockID.length > 8))
+              page.blocks.forEach(b => {
```

Comment:
Please reduce the number of nested code and also add some comments what this code does